src/idlcxx/src/generator.c

```diff
@@ -901,11 +901,11 @@ generate_includes(const idl_pstate_t *pstate, struct generator *generator)
 
   if (generator->uses_array || generator->uses_sequence || generator->uses_bounded_sequence || generator->uses_optional)
   {
-    // ostream cpp
+    // ostream hpp
     // streaming operators for the used std types
     const char *fmt;
     fmt = "#include <org/eclipse/cyclonedds/util/ostream_operators.hpp>\n\n";
-    if (fputs(fmt, generator->impl.handle) < 0)
+    if (fputs(fmt, generator->header.handle) < 0)
       return IDL_RETCODE_NO_MEMORY;
   }
 
```

Comment:
> Hey @michelle-vu19, seems that i missed that case described in #484.
> 
> I put the include of the `ostream_operators.hpp` explicitly into the `cpp` file. Because this file modifies the users std namespace and that is dangerous. So i chose to encapsulate it in the cpp files.
> 
> I strongly recommend to keep it in the cpp files.
> 
> My suggestion is: Iterate all the included idl files and [check if they are using std types](https://github.com/eclipse-cyclonedds/cyclonedds-cxx/blob/a5898dbafdd0da1ec9b19cd55ae31ea4c493b0fb/src/idlcxx/src/generator.c#L902) and include the `ostream_operators.hpp` in the cpp files there too.

I agree with @trittsv on this point. Moving the `ostream_operators.hpp` include from the implementation to the header file might give some unintended side effects, especially since the insertion operators are inside `std` namespace.

The solution I suggest might be a little crude, but it works: just always include the `ostream_operators.hpp` in the implementation file. Since the functions included are template functions, if they're not used, no binary code is generated and since they're in the implementation file, they won't cause any contamination of the `std` namespace.

```suggestion
  // ostream cpp
  // streaming operators for the used std types
  static const char *fmt = "#include <org/eclipse/cyclonedds/util/ostream_operators.hpp>\n\n";
  if (fputs(fmt, generator->impl.handle) < 0)
    return IDL_RETCODE_NO_MEMORY;

  return IDL_RETCODE_OK;
```